### ISSUE TITLE
VegaLite erroneously requires data to be specified at the top level

### ISSUE THREAD

**Issue by jakevdp** [vega/ipyvega#54]
The following spec should produce a valid plot, and does when converted to JSON and fed to vega-lite directly:
```python
spec = {'layers': [{'config': {'mark': {'color': '#1f77b4'}},
   'data': {'values': [{'x': 0, 'y': 0}, {'x': 1, 'y': 1}]},
   'encoding': {'x': {'field': 'x', 'type': 'quantitative'},
    'y': {'field': 'y', 'type': 'quantitative'}},
   'mark': 'line'},
  {'config': {'mark': {'color': '#ff7f0e'}},
   'data': {'values': [{'x': 0, 'y': 1}, {'x': 1, 'y': 0}]},
   'encoding': {'x': {'field': 'x', 'type': 'quantitative'},
    'y': {'field': 'y', 'type': 'quantitative'}},
   'mark': 'point'}]}
```

When used with this package, though, it results in an error:
```python
from vega import VegaLite
VegaLite(spec)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-25-9af8065cc04d> in <module>()
      1 from vega import VegaLite
----> 2 VegaLite(spec)

/Users/jakevdp/anaconda/lib/python3.5/site-packages/vega/base.py in __init__(self, spec, data)
     21         """Initialize the visualization object."""
     22         spec = utils.nested_update(copy.deepcopy(self.DEFAULTS), spec)
---> 23         self.spec = self._prepare_spec(spec, data)
     24 
     25     def _prepare_spec(self, spec, data):

/Users/jakevdp/anaconda/lib/python3.5/site-packages/vega/vegalite.py in _prepare_spec(self, spec, data)
     22 
     23     def _prepare_spec(self, spec, data):
---> 24         return prepare_spec(spec, data)
     25 
     26 

/Users/jakevdp/anaconda/lib/python3.5/site-packages/vega/utils.py in prepare_spec(spec, data)
     91         # Data is either passed in spec or error
     92         if 'data' not in spec:
---> 93             raise ValueError('No data provided')
     94     else:
     95         # As a last resort try to pass the data to a DataFrame and use it

ValueError: No data provided
```

Currently the package assumes that data must appear at the top level; with the new Layer options, this is no longer strictly required.

**Comment by jakevdp** [comment]
The problem is in the [``prepare_spec``](https://github.com/vega/ipyvega/blob/master/vega/utils.py#L74) function; this should not be so strict when data is none.

We'll need to decide whether we validate that ``data`` is in sub-components, or simply let the spec pass through unchecked.

**Comment by jakevdp** [comment]
Ah, just saw that @ellisonbg addressed this in #51
